### PR TITLE
Feature/accordion component

### PR DESCRIPTION
**Pull request by dfrico**
image

Accordion component for the bioclimatic variables page.

### CODE REVIEW

**Comment by aabdaab** [comment]
checking ☝️

File: src/components/accordion/styles.scss [hunk 1]
```diff
@@ -0,0 +1,19 @@
+@import '~styles/settings';
+
+.c-accordion {
+  .accordion-item {
+    background: #E8E8E8;
```
Comment (aabdaab) [inline src/components/accordion/styles.scss]: put into `settings` color variables

File: src/components/accordion/component.jsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      <ul>
+        {items.map(item => (
+          <li key={item.title} className="accordion-item">
+            <button
+              onClick={() => (activeItem === item ? setItem(null) : setItem(item))}
+              className="accordion-title"
+            >
+              {item.title}
+            </button>
+            {activeItem === item && <div className="accordion-content">{item.content}</div>}
```
Comment (aabdaab) [inline src/components/accordion/component.jsx]: we can make a function out of it as it's duplicated in two places, `isActive` or sth

**Review by aabdaab — CHANGES_REQUESTED** [review]
Good job! Check the comments below about the code. There are a couple of things missing from the design as well:
1) We're missing the arrow in the accordion row - check snapshot below
image
2) The background-color of accordion should be white and grey on hover:
image

**Comment by aabdaab** [comment]
Another thing I thought about, it can be left for a different PR. We should be consistent throughout the platform and persist the opened item in the url, I added some util functions in the filters navbar PR

**Comment by aabdaab** [comment]
Hey! I'm posting here the general comments from clicking around the app:
1. On the design, it's indicated that on hovering the accordion row, the chart changes color to pink:
image
It might also be good to have a `cursor: pointer` when hovering over the mini chart 💹
2. We need to tweak the y axis annotation;
**Design**
image
**Now**
image
- The font is different: in the design we've got `Anonymous Pro Regular` and the size of the font is smaller
- The unit (C) should be only on the top of the y-axis 
3. The title of the accordion should be left-aligned with the chart:
**Design**
Screenshot 2019-10-25 at 15 09 30

**Now**
Screenshot 2019-10-25 at 15 08 31

4. The `background-color` of the whole accordion should be white.
5. There is no arrow on the design when having an open accordion item but I think it's OK and more intuitive to have it there, so I would leave it but we can double-check with @fannycc 👍 
image
6. We have a gap between y-axis and the area chart that is not existent in the design:
**Design**
image
**Now**
image
7. The line chart should be a bit thicker, in the design it's 3px. Also, the "dots" indicating the domain values (years) on the line chart should be always visible. 
image
8. We need to make tooltips consistent but I think it's for a different PR as in the species summary chart we would need to do it as well. I will make a PT task for that 👍 Or should we add it to the list pinned on the slack channel? Let me know, I'm happy with both.
**Design**
image
9. The vertical line on y axis is non-existent in the design:
image

I think the idea to feed the real data to the accordion chart is for a different PR, right? It is this item from our slack list?
image
Great job overall, I'm gonna check the code now :)

File: src/components/accordion/styles.scss [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        background-color: rgba($color-gray-1, 0.1);
+      }
+
+      .accordion-minichart{
+        width: 70px;
+        height: 25px;
+      }
+      .expand-icon {
+        // width: 12px;
+        // height: 12px;
```
Comment (aabdaab) [inline src/components/accordion/styles.scss]: Let's remove the commented code

File: src/components/accordion/styles.scss [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+
+    .accordion-content {
+      padding: 6px 18px 30px;
+    }
+
+    &.__open {
+      .accordion-title {
+        font-size: 30px;
```
Comment (aabdaab) [inline src/components/accordion/styles.scss:55]: I think we also need to update the typography:
image
`30px` would stand for the `$h2` typography and now in the settings file it's 32: `$font-size-h2: 32px;`
We could create a `typography` stylesheet and put all our styles for typography there. In Half-Earth we used [sass's placeholder selectors](https://sass-lang.com/documentation/at-rules/extend#placeholder-selectors) and then extended them in specific component style files. For example:
```
// typography.scss
%h1 {
  font-weight: bold;
  font-family: $font-family-1;
  line-height: 80px;
  font-size: 64px;
}
```
and then in the `.accordion-title`:
```
// accordion/styles.scss
...
.accordion-title {
   @extend %h1;
   ... some other things needed for that class ...
}
```

What do you think of that approach?

**Review by aabdaab — COMMENTED** [review]
I left some comment about the typography, the rest of the code looks good 💪 ✊

**Comment by kfilip94** [comment]
:eyes: checking

**Comment by kfilip94** [comment]
@blayhem Thank you for your changes :+1:  I'll take another look tomorrow at your PR but I found a few things that should be done:

- Following Marta's comments above:
  - `"The background-color of accordion should be white and grey on hover:"`

67292158-8dc42280-f4da-11e9-845f-325da9619ad8
  - `"We have a gap between y-axis and the area chart that is not existent in the design:"`
image You need to check Recharts documentation, but I'm not sure if this can be easily done :/

   - `"Another thing I thought about, it can be left for a different PR. We should be consistent throughout the platform and persist the opened item in the url, I added some util functions in the filters navbar PR"` - it'll be done in a new PR?

- Something went wrong here, check x-axis ;)
image


- The y-axis label doesn't match the design:

image
image
One solution could be adding the ` <Label />` inside the `<YAxis> ... <Label /> ... </YAxis>`, check documentation [here](http://recharts.org/en-US/api/Label).
image

After adding this to the `chart/component.jsx`:
image

Please notice that I added `tickSize={0}` to the `<YAxis>` component - it removes this ugly thicker line next to each tick on the y-axis.

Rember about tweaking `CustomTick` component  and adding `fill="#222222"` and `dx={-16}` to the `<text/>` - it allows to have bigger padding between a tick and the chart:
image

And the most important, please double check it with the line chart on the `Species summary` page, because these changes can mess up a little with the styles of the species chart. So the best approach would be to include these changes in the config objects that you already use.

**Pull request merged**